CONVENTION.md

```diff
@@ -0,0 +1,453 @@
+| `util.py` | Like widgets, but non-graphical. Standalone timers, text processing, threading or delay functions.
+| `delegates.py` | If relevant; these create a dependency between model and view, which typically only communicate through the controller.
+
+<br>
+
+#### Structure I - View
+
+This involves the `def __init__()` in `view.py`, the remaining methods being signal and event handlers.
+
+```py
+class Window(QtWidgets.QMainWindow):
+	title = "My Window"
+
+    def __init__(self, model, parent=None):
+    	super(Window, self).__init__(parent)
+    	self.setWindowTitle(self.title)
+        self.setWindowIcon(QtGui.QIcon(...))
+        self.setAttribute(QtCore.Qt.WA_StyledBackground)
+
+    	pages = {
+    		"home": QtWidgets.QWidget(),
+    	}
+
+    	panels = {
+    		"header": QtWidgets.QWidget(),
+    		"body": QtWidgets.QWidget(),
+    		"footer": QtWidgets.QWidget(),
+    		"sidebar": QtWidgets.QWidget(),
+    	}
+
+    	widgets = {
+    		"pages": QtWidgets.QStackedWidget(),
+    		"logo": QtWidgets.QWidget(),
+    		"okBtn": QtWidgets.QPushButton("Ok"),
+    		"cancelBtn": QtWidgets.QPushButton("Cancel"),
+    		"resetBtn": QtWidgets.QPushButton("Reset"),
+    	}
+
+    	icons = {
+    		"cancelBtn": resources.icon("cancelBtn"),
+    	}
+
+```
+
+**Notes**
+
+- **S1** Widgets MUST are separated into `pages`, `panels` and `widgets`
+	- Pages are the full screen of an application, except statusbar, docks, toolbar and menu. Most apps only have one page, but some - like Pyblish Lite and QML - have more
+	- Panels are subdivisions of a Page
+	- Widgets are self explanatory
+- **S2** Widgets MUST be declared at the top
+- **S3** Widgets MUST have a mixedCase name, used in CSS
+- **S4** `Window` MUST have `title` declared as class attribute, used to dynamically change window title at run-time, e.g. to reflect a selection or state
+- **S5** Like widgets, icons SHOULD be declared up-front
+- **S6** Resource logic and I/O SHOULD be handled separately, for testing
+- **S7** Object names SHOULD be mixedCase, to conform with JavaScript/CSS where they are used
+
+```py
+class Window(QtWidgets.QMainWindow):
+	title = "My Window"
+
+    def __init__(self, model, parent=None):
+    	super(Window, self).__init__(parent)
+
+		...
+
+        for name, widget in itertools.chain(panels.items(),
+                                            widgets.items(),
+                                            pages.items()):
+            # Expose to CSS
+            widget.setObjectName(name)
+
+        	# Support for CSS
+            widget.setAttribute(QtCore.Qt.WA_StyledBackground)
+
+        self.setCentralWidget(widgets["pages"])
+        widgets["pages"].addWidget(pages["home"])
+
+```
+
+**Notes**
+
+- **S10** Every widget MUST be given a unique name
+- **S11** Every widget MUST be stylable with CSS
+- **S12** Central widget MUST be a `QStackedWidget` to facilitate multiple pages 
```

Comment:
This seems like an odd requirement? Small apps shouldn't need to have this, right? Seems to me that it's overly complicating apps that have require a single page only. Why would we force this?

Reply:
> This seems like an odd requirement? Small apps shouldn't need to have this, right? Seems to me that it's overly complicating apps that have require a single page only. Why would we force this?

It's only 1 extra line of code for not having to think about this choice and for encouraging additional pages being added; there are a lot of ways in which to add "pages" and without this it isn't clear which to use. Let's keep it.